### PR TITLE
QA tool for custom data

### PR DESCRIPTION
**Pull request by mozilloid**
# References:

Jira: MNTOR-3357, MNTOR-3424, MNTOR-3425
Figma:

# Description
Separated from mock endpoints, and added complete customization. Admin is able to turn on/off custom and real data, and is able to adjust each field of the custom breach and broker objects.

# Screenshot (if applicable)

Not applicable.

# How to test
- Perform db migrations
- Sign in with an admin account
- Enter the data
- Make sure necessary toggles are turned on

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/api/v1/admin/qa-customs/hibp/route.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (rhelmer) [inline src/app/api/v1/admin/qa-customs/hibp/route.ts:1]: Are these useful in stage/prod or only locally? If they only make sense in certain environments please check `APP_ENV` e.g. https://github.com/mozilla/blurts-server/blob/f56d4aac608990e749c9fd6be957a2b65ccb8f88/src/app/(proper_react)/(redesign)/(authenticated)/admin/dev/page.tsx#L16 so it's non-functional on other environments.

**Review by rhelmer — CHANGES_REQUESTED** [review]

File: src/app/api/v1/admin/qa-customs/hibp/route.ts (hunk 1 above)
Comment (mozilloid) [reply src/app/api/v1/admin/qa-customs/hibp/route.ts:1]: These changes should be broadly applicable, which is why I didn't do env checks

File: src/app/api/mock/resetTestData/reset.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  //modify json file
+}
+
+export function brokerAddOneRep() {
+  //modify json file
+}
+
+export function breachAddeHibp() {
+  //modify json file
+}
```
[the remark this answers is not in the document]
Comment (mozilloid) [reply src/app/api/mock/resetTestData/reset.ts]: yes, these are obsolete. Thanks for catching!

File: src/utils/hibp.ts [hunk 3]
```diff
@@ -307,21 +311,37 @@ async function getBreachesForEmail(
...
+  if (qaToggles) {
+    showCustomBreaches = qaToggles.show_custom_breaches;
+    showRealBreaches = qaToggles.show_real_breaches;
+  }
+
+  const qaBreaches = !showCustomBreaches
+    ? []
+    : await getAllQaCustomBreaches(sha1Prefix);
+  if (!showRealBreaches) return qaBreaches as HibpLikeDbBreach[];
+
```
[the remark this answers is not in the document]
Comment (mozilloid) [reply src/utils/hibp.ts:326]: These lines of code check whether a QA toggles row exists for the current user, and if it does, either show the QA breaches or not.

**Review by rhelmer — APPROVED** [review]

**Pull request merged**